ste/uploader-blobFS.go

```diff
@@ -169,22 +175,44 @@ func (u *blobFSUploader) GenerateUploadFunc(id common.ChunkID, blockIndex int32,
 	})
 }
 
+var flushThreshold int64 = (1024 * 1024 * 1024) * 20 //20GB
+//TODO: Should a flag be present for the threshold?
+var thresholdSetup bool
+
+func setupThreshold(blockSize int64) {
+	//Find the closest multiple of the block size to the flush threshold
+	m := flushThreshold % blockSize
+
```

Comment:
Actually... I just realized something. If you check the most recent email we got from the ADLS Gen 2 team, you'll see that they it depends more on the block count, that the size of data.  So, it would be more correct to set flush threshold to something like blocksize * 7500.   That will have the good side effect of giving us much larger flush sizes on large files (since we use larger block sizes there).  I calculate that this may reduce our number of flush calls by a factor of 35 in some cases!  So probably worth doing.  Sorry I didn't think of it in the earlier review!

Reply:
Good point.

Block size might vary for each file, right? so if we rely on block size to figure out the threshold, then `flushThreshold` should perhaps not be a global variable

Reply:
That's fair. I'll rework some of this to flush on a multiple of blockSize (or the end of the file, whichever comes sooner)